### PR TITLE
Control tmpdir for files

### PR DESCRIPTION
**Pull request by frheault**
This PR is related to Issue #40 

This requires PR #37 to be merged first, I needed the new module trx.io to exist to test some things.

Simply replaces all calls to `tempfile.TemporaryDirectory` by a new function below:
```
def get_trx_tmpdir():
    if os.getenv('TRX_TMPDIR') is not None:
        if os.getenv('TRX_TMPDIR') == 'use_working_dir':
            trx_tmp_dir = os.getcwd()
        else:
            trx_tmp_dir = os.getenv('TRX_TMPDIR')
    else:
        trx_tmp_dir = tempfile.gettempdir()

    return tempfile.TemporaryDirectory(dir=trx_tmp_dir, prefix='trx_')
```

3 options:

1. Same behavior (to /tmp)
2. Specify a new folder
3. Use the working directory

I will add a check if the folder does not exist. But should I create it myself?

### CODE REVIEW

**Comment by frheault** [comment]
@anibalsolon What do you think?

File: trx/io.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    dipy_available = True
+except ImportError:
+    dipy_available = False
+
+from trx.utils import split_name_with_gz
+
+
+def get_trx_tmpdir():
+    if os.getenv('TRX_TMPDIR') is not None:
+        if os.getenv('TRX_TMPDIR') == 'use_working_dir':
```
Comment (anibalsolon) [inline trx/io.py:20]: @frheault I gather this would be equivalent to `TRX_TMPDIR=.` ?  Is `.` also valid in Windows?

Comment (frheault) [reply trx/io.py:20]: It is equivalent, but I wanted to write it explicitly rather than using only `.`
I have not tried it on windows.

Comment (anibalsolon) [reply trx/io.py:20]: Sounds good!

**Comment by anibalsolon** [comment]
@frheault looks good- I'll follow the implementation on C++

**Comment by arokem** [comment]
When you get a chance, could you please rebase after the #37 merge?

**Comment by arokem** [comment]
LGTM. Still WIP?

**Comment by frheault** [comment]
It would need to be tested on a Mac, I am not sure how it behaves.
I will ask Maxime D. to borrow one next week

**Comment by arokem** [comment]
Oh - I have a mac. Let me run the test suite and let you know.

**Comment by arokem** [comment]
Tests pass on a mac. Good to go?

**Comment by arokem** [comment]
Actually, could you please rebase on your end first? I tried to do this locally and see there are a couple of conflicts that would need resolution.

**Review by arokem — APPROVED** [review]

**Pull request merged**